### PR TITLE
feature/28_amenities

### PR DESCRIPTION
**Pull request by zilola08**
Added amenities module, controller and service.

### CODE REVIEW

**Review by Zach13131 — APPROVED** [review]
approved

**Review by mirsadikov — APPROVED** [review]

**Comment by zilola08** [comment]
Added swagger to amenities module

**Comment by zilola08** [comment]
Fixed merge conflicts

File: src/amenities/amenities.controller.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+@Controller('amenities')
+export class AmenitiesController {
+  constructor(private amenitiesService: AmenitiesService) {}
+
+  @Get('get/:id')
+  @ApiBearerAuth('JWT-auth')
+  @ApiOperation({ summary: 'GET AMENITIES LIST' })
+  @ApiUnauthorizedResponse()
+  @ApiResponse({
+    status: 201,
```
Comment (bakhromov96) [inline src/amenities/amenities.controller.ts:27]: use correct status codes

File: src/amenities/amenities.controller.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+import { ResponseDto } from './dto/amenititesResponse.dto';
+import { ErrorDto } from './dto/error.dto';
+
+@ApiTags('AMENITIES')
+@UseGuards(UserGuard)
+@Controller('amenities')
+export class AmenitiesController {
+  constructor(private amenitiesService: AmenitiesService) {}
+
+  @Get('get/:id')
```
Comment (bakhromov96) [inline src/amenities/amenities.controller.ts:22]: you can use just /:id

File: src/amenities/amenities.controller.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  @ApiResponse({
+    status: 404,
+    description: 'Incorrect apartment id',
+    type: ErrorDto,
+  })
+  getAmenities(@Param('id') id: string) {
+    return this.amenitiesService.getAmenities(id);
+  }
+
+  @Put('update/:id')
```
Comment (bakhromov96) [inline src/amenities/amenities.controller.ts:40]: same here

**Review by bakhromov96 — CHANGES_REQUESTED** [review]
I didn't put comment, because same comment apply for whole file.

File: prisma/migrations/20231204121925_update_amenities_table/migration.sql [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+  - You are about to drop the column `numberOfRooms` on the `Amenity` table. All the data in the column will be lost.
+  - You are about to drop the column `squareMeters` on the `Amenity` table. All the data in the column will be lost.
+  - A unique constraint covering the columns `[accommodationId]` on the table `Amenity` will be added. If there are existing duplicate values, this will fail.
+
+*/
+-- AlterTable
+ALTER TABLE "Amenity" DROP COLUMN "numberOfRooms",
+DROP COLUMN "squareMeters",
+ADD COLUMN     "airConditioning" BOOLEAN NOT NULL DEFAULT false,
```
[the remark this answers is not in the document]
Comment (zilola08) [reply prisma/migrations/20231204121925_update_amenities_table/migration.sql:11]: renamed all boolean columns as per example

File: src/amenities/amenities.controller.ts (hunk 1 above)
Comment (zilola08) [reply src/amenities/amenities.controller.ts]: corrected the status codes

File: src/amenities/amenities.controller.ts (hunk 2 above)
Comment (zilola08) [reply src/amenities/amenities.controller.ts]: corrected the endpoints

File: src/amenities/amenities.controller.ts (hunk 3 above)
Comment (zilola08) [reply src/amenities/amenities.controller.ts]: corrected the endpoints

File: prisma/schema.prisma [hunk 5]
```diff
@@ -62,11 +62,25 @@ model Accommodation {
 }
 
 model Amenity {
-  id              String        @id @default(uuid())
-  wifi            Boolean
-  parking         Boolean
+  id               String  @id @default(uuid())
+  wifi             Boolean @default(false)
```
[the remark this answers is not in the document]
Comment (zilola08) [reply prisma/schema.prisma]: renamed all boolean columns

**Comment by zilola08** [comment]
made following changes as per comments:

- renamed all boolean columns;
- corrected status codes;
- corrected endpoints;
- corrected error messages and added custom errors;
- adjusted error-handling as per new global error filter;

Could you review the changes, please

File: src/amenities/amenities.service.ts [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+      const updatedAmenities = await this.prisma.amenity.update({
+        where: {
+          accommodationId: id,
+        },
+        data: { ...dto },
+      });
+      return { message: 'Success updating amenities', data: updatedAmenities };
+    } catch (error) {
+      if (error instanceof PrismaClientKnownRequestError) {
+        if (error.code === 'P2025') {
```
Comment (gitkoshelew) [inline src/amenities/amenities.service.ts:78]: add, please Enum for such numbers.  Or maybe it exists somewhere in prisma

**Review by gitkoshelew — APPROVED** [review]

File: src/amenities/amenities.service.ts (hunk 6 above)
Comment (mirsadikov) [reply src/amenities/amenities.service.ts:78]: Enums now created in another PR

Comment (Zach13131) [reply src/amenities/amenities.service.ts:78]: Enums file already exists in develop branch

**Review by bakhromov96 — APPROVED** [review]
Good job!

File: src/amenities/amenities.service.ts (hunk 6 above)
Comment (zilola08) [reply src/amenities/amenities.service.ts]: added enums exclusively for prisma codes

**Pull request merged**